packages/core/src/main/Account.ts

```diff
@@ -389,38 +378,41 @@ class Account extends EventEmitter {
             return this.apiClient.config.store
               .purge()
               .then(() => this.apiClient.init(loginData.persist ? ClientType.PERMANENT : ClientType.TEMPORARY))
-              .then(() => this.service!.client.register(loginData, clientInfo))
-              .then((client: RegisteredClient) => (registeredClient = client))
-              .then(() => {
-                this.logger.info('Client is created');
-                this.apiClient.context!.clientId = registeredClient.id;
-                this.service!.conversation.setClientID(registeredClient.id);
-                return this.service!.notification.initializeNotificationStream(registeredClient.id);
-              })
-              .then(() => this.service!.client.synchronizeClients())
-              .then(() => {
-                return {isNewClient: true, localClient: registeredClient};
-              });
+              .then(() => this.registerClient(loginData, clientInfo));
           }
           this.logger.info('Last client was permanent - Deleting cryptograpy stores');
-          return this.service!.cryptography.deleteCryptographyStores()
-            .then(() => this.service!.client.register(loginData, clientInfo))
-            .then((client: RegisteredClient) => (registeredClient = client))
-            .then(() => {
-              this.logger.info('Client is created');
-              this.apiClient.context!.clientId = registeredClient.id;
-              this.service!.conversation.setClientID(registeredClient.id);
-              return this.service!.notification.initializeNotificationStream(registeredClient.id);
-            })
-            .then(() => this.service!.client.synchronizeClients())
-            .then(() => {
-              return {isNewClient: true, localClient: registeredClient};
-            });
+          return this.service!.cryptography.deleteCryptographyStores().then(() =>
+            this.registerClient(loginData, clientInfo)
+          );
         }
         throw error;
       });
   }
 
+  private registerClient(
+    loginData: LoginData,
+    clientInfo?: ClientInfo
+  ): Promise<{isNewClient: boolean; localClient: RegisteredClient}> {
+    this.logger.info('registerClient');
+    if (!this.service) {
+      throw new Error('Services are not set.');
+    }
+    let registeredClient: RegisteredClient;
+
+    return this.service!.client.register(loginData, clientInfo)
+      .then((client: RegisteredClient) => (registeredClient = client))
+      .then(() => {
+        this.logger.info('Client is created');
+        this.apiClient.context!.clientId = registeredClient.id;
+        this.service!.conversation.setClientID(registeredClient.id);
+        return this.service!.notification.initializeNotificationStream(registeredClient.id);
+      })
+      .then(() => this.service!.client.synchronizeClients())
+      .then(() => {
+        return {isNewClient: true, localClient: registeredClient};
+      });
```

Comment:
no need for three lines: `.then(() => ({isNewClient: true, localClient: registeredClient}));`